DemoApp/DemoAppCoordinator.swift

```diff
@@ -128,22 +128,44 @@ final class DemoAppCoordinator: NSObject, UNUserNotificationCenterDelegate {
         let controller: ChatChannelListController
         
         let connectCompletion: (Error?) -> Void = { [weak self] error in
+            guard let self = self else { return }
             if let error = error {
-                log.error("connecting the user failed \(error)")
-                if let self = self {
+                var errorTitle: String?
+                var errorMessage: String?
+                if let clientError = error as? ClientError,
+                   let underlyingError = clientError.underlyingError as? ClientError,
+                   let engineError = underlyingError.underlyingError as? WebSocketEngineError {
```

Comment:
What if we expose just `Error.isInternetOfflineError`?

Reply:
Yeah, but exposing this is a bit scary:
```swift
                if let clientError = error as? ClientError,
                   let underlyingError = clientError.underlyingError as? ClientError,
                   let engineError = underlyingError.underlyingError as? WebSocketEngineError,
                   engineError.code == WebSocketEngineError.offlineCode ...  {
```
Can we maybe add `ClientError.isInternetOfflineError: Bool` for now?

Reply:
I think it makes sense to provide the helper suggested by @evsaev, otherwise, it is very difficult to get to the `offlineCode` error.